Clarify `String` definition (#2044)
google/comprehensive-rust: src/references/strings.md

We can now understand the two string types in Rust:

- `&str` is a slice of UTF-8 encoded bytes, similar to `&[u8]`.

> - `String` has ownership over its heap-allocated buffer of UTF-8 encoded bytes.

Reviewer: Now that I see the two bullet points together, I feel it would be good for them to have a similar structure: "Foo _is_ X, similar to Y".

So what about this:

```suggestion
- `String` is an owned buffer of UTF-8 encoded bytes, similar to `Vec<T>`.
```

Author: It is ok for me but I want to make a suggestion. Then the desired one can be selected.

```suggestion
- `String` is a type that has ownership over its heap-allocated buffer of UTF-8 encoded bytes.
```

Reviewer: Since we say "the two string types in Rust" just above, I think adding "is a type" here isn't necessary?

Author: Yes you are right. I missed this.
